### PR TITLE
Mock file system in tests

### PR DESCRIPTION
**Pull request by andrewtraviss**
I need to add some more tests to this, but what is done is ready for review.

### CODE REVIEW

File: src/commands/diff/index.ts [hunk 1]
```diff
@@ -79,6 +79,7 @@ export default class Diff extends Base {
     useHTML = false
 
     public async run(): Promise<void> {
+        console.error({cwd: process.cwd()})
```
Comment (laurawarr) [inline src/commands/diff/index.ts:82]: should this still be here?

Comment (andrewtraviss) [reply src/commands/diff/index.ts:82]: Nope

File: src/commands/login/sso.ts [hunk 2]
```diff
@@ -11,8 +10,11 @@ export default class LoginSSO extends AuthCommand {
 
     public async run(): Promise<void> {
         const ssoAuth = new SSOAuth(this.writer)
-        this.token = await ssoAuth.getAccessToken()
-        storeAccessToken(this.token, this.authPath)
+        const accessToken = await ssoAuth.getAccessToken()
+        this.token = accessToken
```
Comment (suthar26) [inline src/commands/login/sso.ts:14]: is this being used anywhere, I see that `authConfig.sso.accessToken` is being used instead

File: src/commands/repo/init.ts [hunk 3]
```diff
@@ -10,15 +9,18 @@ export default class InitRepo extends AuthCommand {
...
         }
 
-        this.repoConfig = await this.updateRepoConfig({})
+        this.dvcConfig.updateRepoConfig({})
 
         const ssoAuth = new SSOAuth(this.writer)
-        this.token = await ssoAuth.getAccessToken()
-        storeAccessToken(this.token, this.authPath)
+        const accessToken = await ssoAuth.getAccessToken()
+        this.token = accessToken
```
Comment (suthar26) [inline src/commands/repo/init.ts:20]: same here unless if I am missing anything

**Review by suthar26 — CHANGES_REQUESTED** [review]

File: src/commands/login/sso.ts (hunk 2 above)
Comment (andrewtraviss) [reply src/commands/login/sso.ts:14]: All of the other commands set and use this.token, and this ensures that bugs will never be introduced in the case that someone assumes that this is true for all commands.

**Review by laurawarr — APPROVED** [review]

**Review by suthar26 — APPROVED** [review]

**Pull request closed without merging**